src/clarity-angular/emphasis/alert/alerts.spec.ts

```diff
@@ -0,0 +1,259 @@
+         ]);
+         service.manage(queryList);
+         component = new Alerts(service);
+      });
+
+      it("knows the current alert", function() {
+         expect(component.currentAlert).toEqual(this.alert);
+      });
+   });
+
+   describe("Template API", function() {
+
+      let fixture: ComponentFixture<any>;
+
+      beforeEach(function() {
+         this.create = <T>(componentType: Type<T>) => {
+            TestBed.configureTestingModule({
+               imports: [ClrEmphasisModule],
+               declarations: [componentType]
+            });
+
+            fixture = TestBed.createComponent(componentType);
+            // fixture.detectChanges();
+         };
+      });
+
+      afterEach(() => {
+         fixture.destroy();
+      });
+
+      it("offers two way binding on the alert index", function() {
+         this.create(TestComponent);
+         const testComponent = fixture.componentInstance;
+         fixture.detectChanges();
+         testComponent.currentAlertIndex = 0;
+         fixture.detectChanges();
+         expect(testComponent.alertsInstance.currentAlertIndex).toEqual(0);
+         testComponent.alertsInstance.currentAlertIndex = 1;
+         fixture.detectChanges();
+         expect(testComponent.currentAlertIndex).toEqual(1);
+      });
+
+      // TODO - Fix
+      it("offers two way binding on the alert instance", fakeAsync(function() {
+          this.create(TestAlertInstance);
+          fixture.detectChanges();
+
+          let clarityComponent = fixture.componentInstance.alertsInstance;
+          let testComponent = fixture.componentInstance;
+
+          expect(testComponent.alertInstances.length).toBe(2);
+          expect(testComponent.alertInstances.first).toBeTruthy();
+          expect(testComponent.alertInstances.last).toBeTruthy();
+
+          testComponent.currentAlert = testComponent.alertInstances.first;
+          fixture.detectChanges();
+          expect(clarityComponent.currentAlert).toEqual(testComponent.currentAlert);
+
+          clarityComponent.currentAlert = testComponent.alertInstances.last;
+          // TODO: figure out why this makes the test hang
+          // fixture.detectChanges();
+          // expect(testComponent.currentAlert).toEqual(clarityComponent.currentAlert);
+      }));
+   });
+
+   describe("View basics", function() {
+
+      let fixture: ComponentFixture<TestComponent>;
+      let compiled: any;
+
+      beforeEach(() => {
+         TestBed.configureTestingModule({
+            imports: [ClrEmphasisModule],
+            declarations: [TestComponent]
+         });
+
+         fixture = TestBed.createComponent(TestComponent);
+         fixture.detectChanges();
+         compiled = fixture.nativeElement;
+      });
+
+      afterEach(() => {
+         fixture.destroy();
+      });
+
+      it("projects content based on clrCurrentAlertIndex", function() {
+         expect(compiled.textContent).toMatch(/This is the second alert/);
```

Comment:
Why is it displaying the second alert here? Everything seems to be initialized with the `currentAlertIndex` being 0, so I don't understand why the second alert displays on initialization.

Reply:
Bad test.  The DOM always contains all the alerts.  I updated the test to check for visibility.  Perhaps a candidate for clrIfOpen in the future?